### PR TITLE
Add in openjdk/maven and update conda-build version

### PR DESCRIPTION
**Pull request by revans2**
This adds openjdk and maven in response to review comments on https://github.com/rapidsai/cudf/pull/2132 and updates the conda-build version because of #30

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -24,9 +24,11 @@ ARG LIBSTDCXX_NG_VERSION=7.3.0
 ARG TINI_VERSION=v0.18.0
 ARG HASH_JOIN=ON
 ARG CONDA_VERSION=4.6.14
-ARG CONDA_BUILD_VERSION=3.18.3
+ARG CONDA_BUILD_VERSION=3.18.5
```
Comment (kkraus14) [inline Dockerfile]: Don't do this, there's a bug in 3.18.5 that breaks build dependencies.

File: Dockerfile.centos7 [hunk 2]
```diff
@@ -24,7 +24,7 @@ ARG LIBGCC_NG_VERSION=7.3.0
 ARG LIBGFORTRAIN_NG_VERSION=7.3.0
 ARG LIBSTDCXX_NG_VERSION=7.3.0
 ARG CONDA_VERSION=4.6.14
-ARG CONDA_BUILD_VERSION=3.18.3
+ARG CONDA_BUILD_VERSION=3.18.5
```
Comment (kkraus14) [inline Dockerfile.centos7]:
```suggestion
ARG CONDA_BUILD_VERSION=3.18.3
```

**Comment by revans2** [comment]
@kkraus14 I made your suggested change, but I have no way to test the change because of #30 I am not a conda expert, but is there something missing that I need to change in my environment to make this work?

**Comment by revans2** [comment]
@kkraus14 it has been a week and I really would like to get moving on CI for java in CUDF so we can fix breaking issues before they happen.  Please let me know if you have any other issues or what the next steps are to get this in.

**Pull request closed without merging**